targets/lenovo-x1-carbon.nix

```diff
@@ -466,6 +474,58 @@
                     ];
                     borderColor = "#337aff";
                   }
+                  {
+                    name = "element";
+                    packages = [pkgs.element-desktop pkgs.gpsd pkgs.element-gps];
+                    macAddress = "02:00:00:03:08:01";
+                    ramMb = 4096;
+                    cores = 1;
+                    extraModules = [
+                      {
+                        services.gpsd = {
```

Comment:
It feels a bit like hack to passthrough the GPS device and run gpsd in the same Virtual Machine than the element itself is running. Maybe we can let this go for now? But have you investigated the possibility of running gpsd in separate VM? What would it need for communication between the VMs? A Socket?

Maybe some TODO-line would be nice for future.

Also I generally don't like putting more stuff into this file, as we are trying to refactor all this stuff away from these target files.

Reply:
Also these services are currently running as root, it would be nicer to create gpsd and element-gps users for these services

Reply:
Here is a very good example for writing a service module: https://github.com/tiiuae/ghaf-caml-crush/blob/main/modules/services/caml-crush/default.nix

Reply:
Actually I misread, the gpsd is fine (https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/misc/gpsd.nix#L113-L122), but proper service would be nice for the element-gps below.

Reply:
> Does it make sense to passthrough the GPS device and run gpsd in the same Virtual Machine than the element itself is running. Maybe we can let this go for now? But have you investigated the possibility of running gpsd in separate VM? What would it need for communication between the VMs? A Socket?
> 
> Maybe some TODO-line would be nice for future.
> 
> Also I generally don't like putting more stuff into this file, as we are trying to refactor all this stuff away from these target files.

I was thinking that kind of solution, but VM for each daemon could be also quite heavy solution?

Reply:
Yeah I was already reading from gpsd source code that it can drop privileges if GPSD_USER and GPSD_GROUP are defined build-time: https://gitlab.com/gpsd/gpsd/-/blob/master/gpsd/gpsd.c?ref_type=heads#L2598-2666

I'm wondering does nixpkgs set these properly to "gpsd"

Reply:
Apparently nixpkgs sets gpsdUser to "gpsd" and gpsdGroup to "dialout" by default:

https://github.com/NixOS/nixpkgs/blob/b0c816c86585ec236c15e668e3d4562d82e183b6/pkgs/servers/gpsd/default.nix#L30

Reply:
So I guess regarding gpsd we are fine.

Reply:
Ok. No changes here.